cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/cdap/CDAPTransactions.java

```diff
@@ -90,29 +97,27 @@ public int getNumCommittedChangeSets() {
   }
 
   @Override
-  public long getVisibilityUpperBound() {
-    return visibilityUpperBound;
-  }
-
-  @Override
   public void collect() throws Exception {
     numInvalidTx = txClient.getInvalidSize();
-    TransactionSnapshot txSnapshot;
-    try (InputStream inputStream = txClient.getSnapshotInputStream()) {
-      txSnapshot = codecProvider.decode(inputStream);
-    }
-    snapshotTime = txSnapshot.getTimestamp();
-    readPointer = txSnapshot.getReadPointer();
-    writePointer = txSnapshot.getWritePointer();
-    numInProgressTx = txSnapshot.getInProgress().size();
-    numCommittingChangeSets = 0;
-    for (Set<ChangeId> changeIds : txSnapshot.getCommittingChangeSets().values()) {
-      numCommittingChangeSets += changeIds.size();
-    }
-    numCommittedChangeSets = 0;
-    for (Set<ChangeId> changeIds : txSnapshot.getCommittedChangeSets().values()) {
-      numCommittedChangeSets += changeIds.size();
+
+    Collection<MetricTimeSeries> collection =
+      metricStore.query(new MetricDataQuery(0, 0, Integer.MAX_VALUE, Integer.MAX_VALUE, METRICS,
+                                            Constants.Metrics.TRANSACTION_MANAGER_CONTEXT,
+                                            Collections.<String>emptyList(), null));
+    for (MetricTimeSeries metricTimeSeries : collection) {
+      if (metricTimeSeries.getMetricName().equals("system.committing.size")) {
+        numCommittingChangeSets = (int) aggregateMetricValue(metricTimeSeries);
+      }
+      if (metricTimeSeries.getMetricName().equals("system.committed.size")) {
+        numCommittedChangeSets = (int) aggregateMetricValue(metricTimeSeries);
+      }
     }
-    visibilityUpperBound = txSnapshot.getVisibilityUpperBound();
+
+    Transaction transaction = txClient.startShort();
+    snapshotTime = System.currentTimeMillis();
```

Comment:
This is not right anymore? Since you are not actually taking a snapshot, isn't this misleading information? The last snapshot time should be the time the last transaction snapshot was taken. Is there a way to get that from the transaction service? If not, let's just remove this?